features/dd-sdk-android-profiling/src/test/kotlin/com/datadog/android/profiling/internal/PerfettoProfilerTest.kt

```diff
@@ -395,6 +396,49 @@ class PerfettoProfilerTest {
         verifyNoInteractions(mockProfilerCallback)
     }
 
+    @Test
+    fun `M not have the same stop signal W start&stop multiple times`() {
+        val inOrder = inOrder(mockService)
+
+        // When
+        // First request
+        testedProfiler.start(mockContext, setOf(fakeInstanceName))
+        val firstStopSignalCaptor = argumentCaptor<CancellationSignal>()
+        testedProfiler.stop(fakeInstanceName)
+
+        // Then
+        inOrder.verify(mockService).requestProfiling(
+            eq(ProfilingManager.PROFILING_TYPE_STACK_SAMPLING),
+            any<Bundle>(),
+            any<String>(),
+            firstStopSignalCaptor.capture(),
+            any(),
+            callbackCaptor.capture()
+        )
+
+        val successResult = mock<ProfilingResult> {
+            on { errorCode } doReturn ProfilingResult.ERROR_NONE
+            on { resultFilePath } doReturn fakePath
+        }
+        callbackCaptor.firstValue.accept(successResult)
+        val captor = argumentCaptor<PerfettoResult>()
+        verify(mockProfilerCallback).onSuccess(captor.capture())
+
+        // Second request
+        val secondStopSignalCaptor = argumentCaptor<CancellationSignal>()
```

Comment:
you can actually reuse the same captor, all captures are recorded and accessible through `allValue`